modules/service/src/main/scala/lucuma/odb/service/ProgramService.scala

```diff
@@ -283,6 +291,16 @@ object ProgramService {
 
   object Statements {
 
+    val selectPid: Query[ProgramReference, Program.Id] =
+      sql"""
+        SELECT
+          c_program_id
+        FROM
+          t_program
+        WHERE
+          c_program_reference = $program_reference
+      """.query(program_id)
```

Comment:
I'd originally thought that wherever a program reference like `G-2025A-0001-Q` would be accepted, we would also accept `G-2025A-0001` (since the proposal reference prefix uniquely identifies the program).  There are several places which will need an update if we do this, including the PID lookup here.

Another would be in the `WhereOptionOrderProgramReference` which has the familiar `EQ`, `NEQ`, `IN`, etc. These would (somehow) have to be changes so that proposal prefix matches would also work.  I'm wondering if we should just require exact matches.  If the proposal is accepted and has a program type like `Q` it becomes required and supplying just `G-2025A-0001` would fail.  What do you think?

Reply:
It makes sense to me to allow the non-suffixed reference, but we probably need to compute and store it so we can index on it. We may already be doing this ... ?

Reply:
I wasn't planning on it but it's trivial to have columns for each.  I'll add that.  There's just the one type `ProgramRerence` for these.  I could add `ProposalReference` but then everywhere that we accept a `ProgramId` or a `ProgramReference` we'd also need to accept a `ProposalReference`.  Maybe that's the most straightforward, easy to understand and easy to implement solution.

Reply:
Having distinct schema types would definitely be the most transparent way to do it, but it's fine with me if we just fall back to proposal reference as an internal thing, if no suffix is specified. My concern was just that doing a prefix match would end up being slow.